homeassistant/components/zha/core/channels/closures.py

```diff
@@ -35,6 +53,63 @@ def attribute_updated(self, attrid, value):
                 f"{self.unique_id}_{SIGNAL_ATTR_UPDATED}", attrid, attr_name, value
             )
 
+    async def async_set_user_code(self, code_slot: int, user_code: str) -> None:
+        """Set the user code for the code slot."""
+
+        set_pin_code = self.__getattr__("set_pin_code")
+        await set_pin_code(
+            *(
+                code_slot - 1,  # start code slots at 1
```

Comment:
how come all of these are shifted by 1? Just curious

Reply:
I shifted by one in part because I want this to be "people" friendly as opposed to "machine" friendly, so start at 1 instead of 0; and in part because z-wave locks start with slot 1, and I wanted to have consistency with them.